src/daft-functions/src/uri/download.rs

```diff
@@ -1,25 +1,28 @@
 use std::sync::Arc;
 
-use common_error::{DaftError, DaftResult};
+use common_error::{ensure, DaftError, DaftResult};
 use common_runtime::get_io_runtime;
 use daft_core::prelude::*;
-use daft_dsl::{functions::ScalarUDF, ExprRef};
+use daft_dsl::{
+    functions::{FunctionArgs, ScalarUDF},
+    ExprRef,
+};
 use daft_io::{get_io_client, Error, IOConfig, IOStatsContext, IOStatsRef};
 use futures::{StreamExt, TryStreamExt};
 use serde::Serialize;
-use snafu::prelude::*;
-
-use crate::InvalidArgumentSnafu;
 
 /// Container for the keyword arguments of `url_download`
 /// ex:
 /// ```text
-/// url_decode(input)
-/// url_decode(input, max_connections=32)
-/// url_decode(input, on_error='raise')
-/// url_decode(input, on_error='null')
-/// url_decode(input, max_connections=32, on_error='raise')
+/// url_download(input)
+/// url_download(input, max_connections=32)
+/// url_download(input, on_error='raise')
+/// url_download(input, on_error='null')
+/// url_download(input, max_connections=32, on_error='raise')
 /// ```
+#[derive(Debug, Clone, Serialize, serde::Deserialize, PartialEq, Eq, Hash)]
+pub struct UrlDownload;
+
 #[derive(Debug, Clone, Serialize, serde::Deserialize, PartialEq, Eq, Hash)]
 pub struct UrlDownloadArgs {
```

Comment:
Is this class still necessary if the udf can track its inputs now?

Reply:
We create unit types like this for registering the function.

Reply:
Wait @universalmind303 I don't see `UrlDownloadArgs` being used anywhere except for a test and a, now unused, `url_download` pyfunction. Are you sure this is needed?

When calling URL download through the function registry, it seems like we bypass the creation of `UrlDownloadArgs`